test/module/irohad/ametsuchi/block_query_test.cpp

```diff
@@ -120,3 +125,70 @@ TEST_F(BlockQueryTest, GetAccountTransactionsNonExistingUser) {
   getNonexistingTxWrapper.subscribe();
   ASSERT_TRUE(getNonexistingTxWrapper.validate());
 }
+
+/**
+ * @given block store with 2 blocks totally containing 3 txs created by
+ * user1@test
+ * AND 1 tx created by user2@test
+ * @when query to get transactions with transactions' hashes
+ * @then queried transactions
+ */
+TEST_F(BlockQueryTest, GetTransactionsExistingTxHashes) {
+  //TODO 15/11/17 motxx - Use EqualList VerificationStrategy
+  {
+    auto wrapper = make_test_subscriber<CallExact>(
+      blocks->getTransactions(tx_hashes), 4);
+    wrapper.subscribe();
+    ASSERT_TRUE(wrapper.validate());
+  }
+  {
+    auto wrapper = make_test_subscriber<CallExact>(
+      blocks->getTransactions({tx_hashes[1], tx_hashes[3]}), 2);
+    wrapper.subscribe();
+    ASSERT_TRUE(wrapper.validate());
+  }
+}
+
+/**
+ * @given block store with 2 blocks totally containing 3 txs created by
+ * user1@test
+ * AND 1 tx created by user2@test
+ * @when query to get transactions with non-existing transaction hashes
+ * @then queried transactions and empty transaction
+ */
+TEST_F(BlockQueryTest, GetTransactionsIncludesNonExistingTxHashes) {
+  iroha::hash256_t invalid_tx_hash_1, invalid_tx_hash_2;
+  invalid_tx_hash_1[0] = 1;
+  invalid_tx_hash_2[0] = 2;
+  {
+    auto wrapper = make_test_subscriber<CallExact>(
+      blocks->getTransactions({invalid_tx_hash_1, invalid_tx_hash_2}), 2);
+    wrapper.subscribe([](auto transaction) {
+      iroha::model::Transaction expected{};
+      EXPECT_EQ(expected, transaction);
+    });
+    ASSERT_TRUE(wrapper.validate());
+  }
+  {
+    // transactions' hashes are empty.
+    auto wrapper = make_test_subscriber<CallExact>(
+      blocks->getTransactions({}), 0);
+    wrapper.subscribe();
+    ASSERT_TRUE(wrapper.validate());
+  }
+  {
+    //TODO 15/11/17 motxx - Use EqualList VerificationStrategy
+    auto wrapper = make_test_subscriber<CallExact>(
+      blocks->getTransactions({invalid_tx_hash_1, tx_hashes[0]}), 2);
+    auto subs_cnt = 0;
```

Comment:
1. Cant' find any variable modification (though [there's if check](https://github.com/hyperledger/iroha/pull/696/files#diff-c855aee60870118507c2cd5177c150eaR185))
2. This var can be static inside the subscribing lambda